### PR TITLE
Fix ESLint warnings, migrate Husky from 4 version to 7, Prettier and ESLint use common rules

### PR DESCRIPTION
**Pull request by OlgaUrentseva**
# Summary

#### A brief description of the pull request.
- fix ESLint warnings
- deny `any` usage
- refactor typings for Form component
- update Husky from 4 version to 7
- Prettier and ESLint use common rules

#### Screenshots in case of UI changes

# Test plan

#### List of steps to manually test introduced functionality:

- Go to [ReviewApp](ReviewAppLink)
- ...

#### What do need to pay more attention to, some implicit changes that influence the app:

- ...

# Review notes

#### While reviewing pull-request (especially when it's your pull-request), please make sure that:

- you named PR close to issue name and linked that issue
- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### CODE REVIEW

File: components/pages/login/components/LoginForm/LoginForm.jsx [hunk 1]
```diff
@@ -34,11 +34,11 @@ const LoginForm = () => {
 
   useEffect(() => {
     if (detailMessage) setSuccess(detailMessage);
-  }, [detailMessage]);
+  }, [detailMessage, setSuccess]);
```
Comment (MaksimVdovin) [inline components/pages/login/components/LoginForm/LoginForm.jsx:37]: are u sure that we need to add function in array of dependence?
it does't have meaning there

File: components/pages/login/components/LoginForm/LoginForm.jsx [hunk 2]
```diff
@@ -34,11 +34,11 @@ const LoginForm = () => {
...
 
   useEffect(() => {
     if (errorMessage) setError(errorMessage);
-  }, [errorMessage]);
+  }, [errorMessage, setError]);
```
Comment (MaksimVdovin) [inline components/pages/login/components/LoginForm/LoginForm.jsx:41]: and the same there

File: components/shared/organisms/Header/UserNavigation.tsx [hunk 3]
```diff
@@ -14,15 +14,15 @@ interface Props extends IUserNavigation {
...
 
   const toggleDropdown = useCallback(() => {
     setIsOpen(!isOpen);
   }, [isOpen]);
 
   useEffect(() => {
-    const handleOutsideClick = (event: any) => {
-      if (wrapperRef?.current && wrapperRef?.current?.contains(event.target)) {
+    const handleOutsideClick = (event: MouseEvent) => {
+      if (wrapperRef?.current && event.target instanceof Node && wrapperRef?.current?.contains(event.target)) {
```
Comment (MaksimVdovin) [inline components/shared/organisms/Header/UserNavigation.tsx:25]: also. i think we can move this function from useEffect body (look toggleDropdown)

and dont need to add dependency in array at useEffect. 

also. can remove toggleDropdown from 40 line.

**Review by nadezhdakharchuk — APPROVED** [review]

File: components/shared/organisms/Header/UserNavigation.tsx (hunk 3 above)
Comment (OlgaUrentseva) [reply components/shared/organisms/Header/UserNavigation.tsx:25]: I don't think its a good idea to move function from useEffect according to https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Comment (MaksimVdovin) [reply components/shared/organisms/Header/UserNavigation.tsx:25]: yeah. thank u! dont need really.

File: components/pages/activity/components/ActivityTable/DataCell.tsx [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
 
 import { StyledCell, Wrap } from './styled';
 
-const DataCell = ({ children }) => {
+const DataCell = ({ children }: { children: ReactNode }) => {
```
Comment (DmitriyFirsov) [inline components/pages/activity/components/ActivityTable/DataCell.tsx:5]: PropsWithChildren<{}>

**Review by DmitriyFirsov — APPROVED** [review]

**Pull request merged**